linux_os/guide/system/software/integrity/crypto/configure_openssl_tls_crypto_policy/ansible/shared.yml

```diff
@@ -3,16 +3,14 @@
 # strategy = restrict
 # complexity = low
 # disruption = low
+{{{ ansible_instantiate_variables("var_system_crypto_policy") }}}
 
-{{{ ansible_set_config_file(
-        msg='Ensure system crypto policy has OpenSSL library using TLS encryption',
-        file='/etc/crypto-policies/back-ends/opensslcnf.config',
-        parameter='MinProtocol',
-        value='TLSv1.2',
-        create='yes',
-        separator=' = ',
-        separator_regex="\s*=\s*",
-        insert_before='^#?MaxProtocol ',
-        prefix_regex="^"
-    )
-}}}
+- name: "{{{ rule_title }}}"
+  lineinfile:
+    path: /etc/crypto-policies/config
+    regexp: '^(?!#)(\S+)$'
+    line: "{{ var_system_crypto_policy }}"
+    create: yes
+
+- name: Verify that Crypto Policy is Set (runtime)
+  command: /usr/bin/update-crypto-policies --set {{ var_system_crypto_policy }}
```

Comment:
@yuumasato doesn't this reset the crypto-policy files?

Reply:
@carlosmmatos Yes, the remediation will set the crypto-policy defined by the profile.

Reply:
I reckon that setting the crypto-policy may not be desired.
So I'll drop the remediation and a recommendation in the rule that the admin needs to look for a crypto-policy that satisfies the rule's requirement.

Reply:
Done in 97d792e, please check it out.

Reply:
Could choice of crypto policy be controlled via an XCCDF variable? It seems like certain profiles OSPP/STIG have very particular choices of profiles (probably, FIPS) and we could use the DEFAULT crypto policy by default.

Reply:
@cipherboy There is a [variable](https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/software/integrity/crypto/var_system_crypto_policy.var) that can be used together with rule [configure_crypto_policy](https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/software/integrity/crypto/configure_crypto_policy/rule.yml). But I don't where you are heading with the question.

Reply:
@yuumasato I think it was trying to address @carlosmmatos's concern :-) He was worried about it resetting crypto-policy, but we already use a variable here and it aligns more or less with the existing rule as you mentioned. I think removing the additional remediation is correct, but I think we're missing an actual dependency from this rule onto `configure_crypto_policy` right? My 2c., but I think we want to make sure the policy rule gets dragged into the profile as well as this rule.